packages/guilded.js/lib/gateway/handler/ForumEventHandler.ts

```diff
@@ -26,28 +22,38 @@ export class ForumEventHandler extends GatewayEventHandler {
         return this.client.emit(constants.clientEvents.FORUM_TOPIC_CREATED, newTopic);
     }
     forumTopicUpdated(data: WSForumTopicUpdated) {
-        const getCachedtopic = this.client.topics.cache.get(data.d.forumTopic.id);
-        if (!getCachedtopic) {
-            const newtopic = new ForumTopic(this.client, data.d.forumTopic);
-            return this.client.emit(constants.clientEvents.FORUM_TOPIC_UPDATED, newtopic, null);
+        const getCachedTopic = this.client.topics.cache.get(data.d.forumTopic.id);
+        if (!getCachedTopic) {
+            const newTopic = new ForumTopic(this.client, data.d.forumTopic);
+            return this.client.emit(constants.clientEvents.FORUM_TOPIC_UPDATED, newTopic, null);
         }
-        const frozenOldtopic = Object.freeze(getCachedtopic._clone());
-        getCachedtopic._update(data.d.forumTopic);
-        return this.client.emit(constants.clientEvents.FORUM_TOPIC_UPDATED, getCachedtopic, frozenOldtopic);
+        const frozenOldTopic = Object.freeze(getCachedTopic._clone());
+        getCachedTopic._update(data.d.forumTopic);
+        return this.client.emit(constants.clientEvents.FORUM_TOPIC_UPDATED, getCachedTopic, frozenOldTopic);
     }
     forumTopicDeleted(data: WSForumTopicDeleted) {
-        const getCachedtopic = this.client.topics.cache.get(data.d.forumTopic.id);
-        getCachedtopic?._update({ _deletedAt: new Date() });
-        return this.client.emit(constants.clientEvents.FORUM_TOPIC_DELETED, getCachedtopic ?? new ForumTopic(this.client, data.d.forumTopic));
+        const getCachedTopic = this.client.topics.cache.get(data.d.forumTopic.id);
+        getCachedTopic?._update({ _deletedAt: new Date() });
+        return this.client.emit(constants.clientEvents.FORUM_TOPIC_DELETED, getCachedTopic ?? new ForumTopic(this.client, data.d.forumTopic));
     }
     forumTopicPinned(data: WSForumTopicPinned) {
-        const getCachedtopic = this.client.topics.cache.get(data.d.forumTopic.id);
-        getCachedtopic?._update({ isPinned: true });
-        return this.client.emit(constants.clientEvents.FORUM_TOPIC_PINNED, getCachedtopic ?? new ForumTopic(this.client, data.d.forumTopic));
+        const getCachedTopic = this.client.topics.cache.get(data.d.forumTopic.id);
+        getCachedTopic?._update({ isPinned: true });
+        return this.client.emit(constants.clientEvents.FORUM_TOPIC_PINNED, getCachedTopic ?? new ForumTopic(this.client, data.d.forumTopic));
     }
     forumTopicUnpinned(data: WSForumTopicUnpinned) {
-        const getCachedtopic = this.client.topics.cache.get(data.d.forumTopic.id);
-        getCachedtopic?._update({ isPinned: false });
-        return this.client.emit(constants.clientEvents.FORUM_TOPIC_UNPINNED, getCachedtopic ?? new ForumTopic(this.client, data.d.forumTopic));
+        const getCachedTopic = this.client.topics.cache.get(data.d.forumTopic.id);
```

Comment:
Should we abstract this kinda logic into a helper function or method so we don't have to duplicate this line in every event? Or maybe we should have a step before this event function is called where we check the cache and pass it through as a parameter in this method.

Reply:
Not sure how a helper function would be useful for this. You'd still need to call that func on every event right

Reply:
Let me rephrase, what if we had a step BEFORE calling methods like forumTopicUpdated where we get the cached structure and pass it as a parameter into forumTopicUpdated.

Reply:
Ah. I feel like in this case the abstraction is unnecessary and maybe even counterproductive since it would create an avoidable inconsistency with how the forum events are handled vs all others (there would be an additional step).

If you think this would be better though, sure I don't mind.